aips/aip-x.md

```diff
@@ -0,0 +1,707 @@
+ * specifications.
+ *
+ * @param input The input to create the SignIn message text from.
+ *
+ * @returns The SignIn message text.
+ */
+function createSignInMessageText(input: AptosSignInMessage["fields"]): string;
+
+/**
+ * Verifies an input SignIn message against expected fields.
+ *
+ * @param input The input to verify the message against.
+ * @param expected The expected message to verify against the input.
+ *
+ * @returns The verification result.
+ */
+function verifySignInMessage(
+  input: AptosSignInInput,
+  expected: AptosSignInMessage | AptosSignInMessage["fields"] | string
+): VerificationResult<AptosSignInMessage["fields"]>;
+
+/**
+ * Verifies outputs from a `signIn` method response against input fields.
+ *
+ * @param input The input to verify the output against.
+ * @param output The output to verify against the input.
+ *
+ * @returns The verification result.
+ */
+function verifySignIn(
+  input: AptosSignInInput,
+  output: {
+    publicKey: PublicKey;
+    signature: Signature;
+    message: string;
+  }
+): VerificationResult<AptosSignInMessage["fields"]>;
+
+/**
+ * Generate a signing message using the Sign in with Aptos signing algorithm.
+ * sha3_256( sha3_256(b"SIGN_IN_WITH_APTOS::" ) || <message> )
+ *
+ * @param message The SIWA message to sign.
+ 
+ * @returns The signing message.
+ */
+function generateSignInSigningMessage(message: string): Uint8Array;
+```
+
+#### @aptos-labs/wallet-standard
+
+The following interfaces would be introduced into the `@aptos-labs/wallet-standard` package.
+
+```ts
+import { AccountInfo, Signature } from "@aptos-labs/wallet-standard";
+
+export type AptosSignInInput = {
+  domain?: string;
+  address?: string;
+  uri?: string;
+  version?: string;
+  statement?: string;
+  nonce?: string;
+  chainId?: string;
+  issuedAt?: string;
+  expirationTime?: string;
+  notBefore?: string;
+  requestId?: string;
+  resources?: string[];
+};
+
+export type AptosSignInMessage = {
+  fields: {
+    domain: string;
+    address: string;
+    uri: string;
+    version: string;
+    statement?: string;
+    nonce?: string;
+    chainId: string;
+    issuedAt?: string;
+    expirationTime?: string;
+    notBefore?: string;
+    requestId?: string;
+    resources?: string[];
+  };
+  plainText: string;
+  signature: Uint8Array;
+  type: string;
+};
+
+export type AptosSignInOutput = {
+  account: AccountInfo;
+  message: AptosSignInMessage;
+  signature: Signature;
+};
+```
+
+### Wallet Implementation
+
+The wallet is responsible for taking `AptosSignInInput` fields defined by an an application and returning a response object with a signature, plain text representation of the message, and account information.
+
+The wallet must bind the `address`, `chainId` , `domain` , `uri` , and `version` fields if not present. If there is a mismatch between these fields and the ones defined by the application, it is important that it is flagged and shown to the user.
+
+#### Overview
```

Comment:
I found this hard to read / understand. Not a TS guy so that may be why. But there may also be too many things here. Do you need all of this? Can anything be left out / simplified?